### PR TITLE
Fix unstable RMSNorm

### PR DESCRIPTION
**Pull request by ZYHowell**
As mentioned in [flax's doc](https://flax.readthedocs.io/en/latest/_modules/flax/linen/normalization.html), with fp16 training, normalization should still use fp32 for stability.
Below is what I observed in evaluating LLaMA with EasyLM. All experiments use the same sample:
`dtype=jnp.float32` or `dtype=jnp.bfloat16`: the loss is 0.12
`dtype=jnp.float16`, without this pr's fix: the loss is 2
`dtype=jnp.float16`, with this pr's fix: the loss is 0.12

### CODE REVIEW

**Comment by young-geng** [comment]
Thanks a lot for the fix! Do you know if this also affects bfloat16?

**Comment by ZYHowell** [comment]
I tested few samples on bf16 with the fix, the loss looks good(still around 0.12 for the same sample as above)
